RenderSystems/GLSupport/src/EGL/Wayland/OgreWaylandEGLWindow.cpp

```diff
@@ -0,0 +1,297 @@
+        }
+    }
+}
+
+void WaylandEGLWindow::windowMovedOrResized()
+{
+    if (mClosed || !mWindow)
+        return;
+
+    int width, height;
+    wl_egl_window_get_attached_size(mWindow, &width, &height);
+    resize(width, height);
+}
+
+void WaylandEGLWindow::create(const String& name, uint width, uint height, bool fullScreen,
+                              const NameValuePairList* miscParams)
+{
+    String title = name;
+    int samples = 0;
+    short frequency = 0;
+    int maxBufferSize(32), minBufferSize(16), maxDepthSize(16), maxStencilSize(0);
+    bool vsync = false;
+    ::EGLContext eglContext = NULL;
+
+    unsigned int vsyncInterval = 1;
+
+    //mNativeDisplay = mGLSupport->getNativeDisplay();
+    mIsFullScreen = fullScreen;
+
+    if (miscParams)
+    {
+        NameValuePairList::const_iterator opt;
+        NameValuePairList::const_iterator end = miscParams->end();
+
+        if ((opt = miscParams->find("currentGLContext")) != end && StringConverter::parseBool(opt->second))
+        {
+            eglContext = eglGetCurrentContext();
+            EGL_CHECK_ERROR
+            if (!eglContext)
+            {
+                OGRE_EXCEPT(Exception::ERR_RENDERINGAPI_ERROR,
+                            "currentGLContext was specified with no current GL context", "EGLWindow::create");
+            }
+
+            mEglDisplay = eglGetCurrentDisplay();
+            EGL_CHECK_ERROR
+        }
+
+
+        if((opt = miscParams->find("maxColourBufferSize")) != end)
+        {
+          maxBufferSize = Ogre::StringConverter::parseInt(opt->second);
+        }
+
+        if((opt = miscParams->find("maxDepthBufferSize")) != end)
+        {
+          maxDepthSize = Ogre::StringConverter::parseInt(opt->second);
+        }
+
+        if((opt = miscParams->find("maxStencilBufferSize")) != end)
+        {
+          maxStencilSize = Ogre::StringConverter::parseInt(opt->second);
+        }
+
+        if((opt = miscParams->find("minColourBufferSize")) != end)
+        {
+          minBufferSize = Ogre::StringConverter::parseInt(opt->second);
+          if (minBufferSize > maxBufferSize) minBufferSize = maxBufferSize;
+        }
+
+        if ((opt = miscParams->find("FSAA")) != end)
+        {
+            samples = StringConverter::parseUnsignedInt(opt->second);
+        }
+
+        if ((opt = miscParams->find("displayFrequency")) != end)
+        {
+            frequency = (short)StringConverter::parseInt(opt->second);
+        }
+
+        if ((opt = miscParams->find("vsync")) != end)
+        {
+            vsync = StringConverter::parseBool(opt->second);
+        }
+
+        if ((opt = miscParams->find("vsyncInterval")) != end)
+            vsyncInterval = StringConverter::parseUnsignedInt(opt->second);
+
+        if ((opt = miscParams->find("gamma")) != end)
+        {
+            mHwGamma = StringConverter::parseBool(opt->second);
+        }
+
+        if ((opt = miscParams->find("title")) != end)
```

Comment:
title is unused now

Reply:
Removed and also changed signature of affected function